equality used to compare floating point numbers (test_bootstrap_alternative)

Values calculated by the bootstrap method are tested at
https://github.com/scipy/scipy/blob/1e3e8c603126f1bf2e8febd567cfd374d169283b/scipy/stats/tests/test_resampling.py#L540
But the test uses equality to compare the actual and desired values. This inevitably fails for floating point numbers, for instance on i386 (debian build tests on scipy v1.11.1)
```
______________________ test_bootstrap_alternative[basic] _______________________

method = 'basic'

    @pytest.mark.parametrize("method", ['basic', 'percentile', 'BCa'])
    def test_bootstrap_alternative(method):
        rng = np.random.default_rng(5894822712842015040)
        dist = stats.norm(loc=2, scale=4)
        data = (dist.rvs(size=(100), random_state=rng),)
    
        config = dict(data=data, statistic=np.std, random_state=rng, axis=-1)
        t = stats.bootstrap(**config, confidence_level=0.9)
    
        config.update(dict(n_resamples=0, bootstrap_result=t))
        l = stats.bootstrap(**config, confidence_level=0.95, alternative='less')
        g = stats.bootstrap(**config, confidence_level=0.95, alternative='greater')
    
>       assert_equal(l.confidence_interval.high, t.confidence_interval.high)
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: 4.6006151751157525
E        DESIRED: 4.600615175115755

config     = {'axis': -1, 'bootstrap_result': BootstrapResult(confidence_interval=ConfidenceInterval(low=3.6279404186700406, high=4...382229, -0.1082399 ,
        0.70475099, 10.78761938,  1.05744104,  8.45953715,  3.85913619]),), 'n_resamples': 0, ...}
data       = (array([ 2.62480303,  0.94487726,  3.91730505,  5.3708377 ,  5.43231996,
        9.19056041,  3.97648906,  3.98515237,...5813,  1.05203055, -0.55382229, -0.1082399 ,
        0.70475099, 10.78761938,  1.05744104,  8.45953715,  3.85913619]),)
dist       = <scipy.stats._distn_infrastructure.rv_continuous_frozen object at 0xd1c16300>
g          = BootstrapResult(confidence_interval=ConfidenceInterval(low=3.627940418670041, high=array(inf)), bootstrap_distribution....90581787, 4.42585593, 4.45999832, ..., 4.03392801, 3.98546718,
       4.18018658]), standard_error=0.2937303808414012)
l          = BootstrapResult(confidence_interval=ConfidenceInterval(low=array(-inf), high=4.6006151751157525), bootstrap_distributi....90581787, 4.42585593, 4.45999832, ..., 4.03392801, 3.98546718,
       4.18018658]), standard_error=0.2937303808414012)
method     = 'basic'
rng        = Generator(PCG64) at 0xD1ED3398
t          = BootstrapResult(confidence_interval=ConfidenceInterval(low=3.6279404186700406, high=4.600615175115755), bootstrap_dist....90581787, 4.42585593, 4.45999832, ..., 4.03392801, 3.98546718,
       4.18018658]), standard_error=0.2937303808414012)

scipy/stats/tests/test_resampling.py:538: AssertionError
```
The same error occurs with other `test_bootstrap_alternative` methods, `[percentile]` and `[BCa]`. Full log at https://buildd.debian.org/status/fetch.php?pkg=scipy&arch=i386&ver=1.11.1-1exp1&stamp=1707086250&raw=0

The values reported in the error log would pass with `assert_allclose` instead of `assert_equal`

--- a/scipy/stats/tests/test_resampling.py
+++ b/scipy/stats/tests/test_resampling.py
@@ -537,8 +537,10 @@ def test_bootstrap_alternative(method):
     l = stats.bootstrap(**config, confidence_level=0.95, alternative='less')
     g = stats.bootstrap(**config, confidence_level=0.95, alternative='greater')
 
-    assert_equal(l.confidence_interval.high, t.confidence_interval.high)
-    assert_equal(g.confidence_interval.low, t.confidence_interval.low)
+    assert_allclose(l.confidence_interval.high, t.confidence_interval.high,
+                    rtol=1e-14)
+    assert_allclose(g.confidence_interval.low, t.confidence_interval.low,
+                    rtol=1e-14)
     assert np.isneginf(l.confidence_interval.low)
     assert np.isposinf(g.confidence_interval.high)
 

